test/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+set(SRCROOT "${PROJECT_SOURCE_DIR}/test/src")
+
+include_directories("${PROJECT_SOURCE_DIR}/include")
+include_directories("${PROJECT_SOURCE_DIR}/extlibs/headers")
+include_directories("${SRCROOT}/TestUtilities")
+
+# System is always built
+SET(SYSTEM_SRC
+    "${SRCROOT}/CatchMain.cpp"
+    "${SRCROOT}/System/Vector2.cpp"
+    "${SRCROOT}/System/Vector3.cpp"
+    "${SRCROOT}/TestUtilities/SystemUtil.hpp"
+    "${SRCROOT}/TestUtilities/SystemUtil.cpp"
+)
+sfml_add_test(systemtest "${SYSTEM_SRC}"  sfml-system)
+
+if(SFML_BUILD_WINDOW)
+    SET(WINDOW_SRC
+        "${SRCROOT}/CatchMain.cpp"
+        "${SRCROOT}/TestUtilities/WindowUtil.hpp"
+        "${SRCROOT}/TestUtilities/WindowUtil.cpp"
+    )
+    sfml_add_test(windowtest "${WINDOW_SRC}" sfml-window)
+endif()
+
+if(SFML_BUILD_GRAPHICS)
+    SET(GRAPHICS_SRC
+        "${SRCROOT}/CatchMain.cpp"
+        "${SRCROOT}/Graphics/Rect.cpp"
+        "${SRCROOT}/TestUtilities/GraphicsUtil.hpp"
+        "${SRCROOT}/TestUtilities/GraphicsUtil.cpp"
+    )
+    sfml_add_test(graphicstest "${GRAPHICS_SRC}" sfml-graphics)
+endif()
+
```

Comment:
I would also add a test target for sfml-network and sfml-audio, for completeness. And so that people who want to contribute don't have to create them.

Reply:
It's not hard to add and we don't want to add stuff just because it hopefully will be added one day, same as we don't add C++ functions because they might be useful someday. 😉